pyogrio/raw.py

```diff
@@ -84,19 +85,36 @@ def read(
             "geometry": "<geometry type>"
         }
     """
+    from_buffer = False
+    if isinstance(path_or_buffer, bytes):
+        from_buffer = True
+        ext = ""
+        if path_or_buffer[:4].startswith(b'PK\x03\x04'):
+            ext = ".zip"
```

Comment:
This is also coming from Fiona (https://github.com/Toblerity/Fiona/blob/c30bb067da91d7c9f71654bc0fd4761ad066ff0a/fiona/collection.py#L688-L695). It's not very generic, but actually supporting zipped buffers for the in-memory reading would be good I think